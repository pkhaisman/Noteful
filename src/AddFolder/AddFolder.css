.AddFolder__form {
    width: 150px;
    margin: 0 auto;
}

.AddFolder__buttons {
    display: flex;
    justify-content: space-between;
    margin-top: 5px;
}

.AddFolder__new-folder-input {
    width: 98%;
}

.AddFolder__cancel {
    border-radius: 5px;
    background-color: #f44336;
    border: none;
    padding: 5px 10px;
    font-weight: bold;
}

.AddFolder__cancel:hover {
    background-color: #db3b2f;
    cursor: pointer;
}

.AddFolder__cancel--color {
    color: white;
}

.AddFolder__submit {
    border-radius: 5px;
    background-color: #4CAF50;
    border: none;
    color: white;
    padding: 5px 10px;
    font-weight: bold;
}

.AddFolder__submit:hover {
    cursor: pointer;
    background-color: rgb(63, 148, 66);
}

button:disabled {
    border-radius: 5px;
    background-color: grey;
    border: none;
    color: white;
    padding: 5px 10px;
    font-weight: bold;
}

button:disabled:hover {
    background-color: grey
}

.error {
    color: #db3b2f
}