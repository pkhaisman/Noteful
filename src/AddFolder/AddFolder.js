import React, {Component} from 'react';
import {Link, withRouter} from 'react-router-dom';
import NotefulContext from '../NotefulContext';
import ValidationError from '../ValidationError'

class AddFolder extends Component {
    static contextType = NotefulContext;

    constructor(props) {
        super(props);
        this.state = {
            newFolder: '',
            formValid: false,
            validationMessage: ''
        }
    }

    updateNewFolder = (newFolder) => {
        this.setState({newFolder}, this.validateForm(newFolder));
    }

    validateForm = (fieldValue) => {
        let errorMessage = this.state.validationMessage;
        let hasError = false;

        fieldValue = fieldValue.trim();
        if (fieldValue.length === 0) {
            errorMessage = 'Input is required';
            hasError = true;
        }
        this.setState({
            formValid: !hasError,
            validationMessage: errorMessage
        });
    }

    handleSubmit = (e) => {
        e.preventDefault();
        const folderName = this.state.newFolder;
        // make a POST request
        fetch('http://localhost:9090/folders', {
            method: 'POST',
            body: JSON.stringify({
                name: folderName

            }),
            headers: {
                'content-type': 'application/json'
            }
        })
            .then(response => {
                if (!response.ok) {
                    throw new Error('An error ocurred in your post request');
                }
                return response.json()
            })
            .then(data => {
                this.setState({
                    newFolder: ''
                });
                if (this.state.formValid) {
                    this.context.handleAddFolder(folderName, data.id);
                    this.props.history.push('/');
                }
            })
            .catch(error => {
                console.log(error);
            })
    }

    render() {
        return (
            <div className='AddFolder'>
                <form onSubmit={e => this.handleSubmit(e)}>
                    <h2>Add a Folder</h2>
                    <div className='input-field'>
                        <label htmlFor='new-folder'></label>
                        <input type='text' className='new-folder-input'
                            name='new-folder' id='new-folder' 
                            onChange={(e) => this.updateNewFolder(e.target.value)} />
                    </div>
                    <div className='buttons'>
                        <Link
                            to='/'
                        >
                            <button>Cancel</button>
                        </Link>
                        <button disabled={!this.state.formValid}>Submit</button>
                    </div>
                </form>
                <ValidationError hasError={!this.state.formValid} message={this.state.validationMessage} />
            </div>
        )
    }
}

export default withRouter(AddFolder);