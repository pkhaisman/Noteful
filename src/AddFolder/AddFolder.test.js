import React from 'react';
import AddFolder from './AddFolder';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('AddFolder Component', () => {

    // does this qualify as a snapshot test?
    it('renders without crashing', () => {
        shallow(<AddFolder />)
    });
    
    it('renders the UI as expected', () => {
        const wrapper = shallow(<AddFolder />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});