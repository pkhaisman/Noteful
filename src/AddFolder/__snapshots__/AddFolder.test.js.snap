// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder Component renders the UI as expected 1`] = `
<div
  className="AddFolder"
>
  <form>
    <h2>
      Add a Folder
    </h2>
    <div
      className="input-field"
    >
      <label
        htmlFor="new-folder"
      />
      <input
        className="new-folder-input"
        id="new-folder"
        name="new-folder"
        type="text"
      />
    </div>
    <div
      className="buttons"
    >
      <button>
        Cancel
      </button>
      <button>
        Sumbit
      </button>
    </div>
  </form>
</div>
`;
